middleware/verifyUser.js

```diff
@@ -1,39 +1,47 @@
+import { compareSync } from 'bcrypt';
 import { User } from '../models';
 
+let userWantsToLogin = false;
+
 export default (req, res, next) => {
-  const {
-    body,
-    user,
-  } = req;
+  const { body, user } = req;
   const options = {};
   if (user) {
     options.fieldName = 'id';
     options.fieldValue = user.id;
   } else if (body) {
     options.fieldName = 'email';
     options.fieldValue = body.email;
+    userWantsToLogin = true;
```

Comment:
@chingsley what does this do? Also, can you think of a better variable name?

Reply:
I use it to differentiate when a user is trying to log in, from when the user is already logged in and is trying to perform another action that requires authentication such view profile. So I implemented this middleware in such a way that my team members, e.g the person that implemented the feature to view user profile can also call the function. So further down in that function, there is a place where I compare password. That can only happen when a user is trying to log in. If I tried to compare the password when a user is performing other actions, such as trying to view their profile, an error will be thrown because a password is not provided then.